### ISSUE TITLE
Navigation back looses forward navigation parameters from backstack [UWP, Win10]

### ISSUE THREAD

**Issue by FrogsLegs** [PrismLibrary/Prism#1728]
### Description
Navigation parameters are lost when navigating back via `NavigationService.GoBackAsync();`. 
More details over on this [StackOverflow](https://stackoverflow.com/questions/55040487/prism-7-uwp-navigation-back-parameters-missing) post. 

### Steps to Reproduce

1. Create a new UWP App, add three Views / ViewModels
2. Navigate from View1 to View2 with a parameter
3. Navigate from View2 to View3
4. Navigate back by calling `GoBackAsync`
5. 'OnNavigatedToAsync' in ViewModel2 is called as expected, but the parameter initially passed when navigating from View1 to View2 is lost.

### Expected Behavior
In step 5 above the parameter passed from View1 to View2 should be present, as it is in a non-prism UWP app.

### Actual Behavior
Parameters are lost when navigating back

### Basic Information

- Version with issue: v7.2.0.1038-pre
- Last known good version: N/A
- IDE: VS2017

### Reproduction Link
[PrismNavBackExample.zip](https://github.com/PrismLibrary/Prism/files/2945590/PrismNavBackExample.zip)

**Comment by FrogsLegs** [comment]
I'm trying to clone this repo to step through the behaviour in code, having a nightmare trying to resolve some of the older SDKs (uap10.0.15063) and (uap10.0.16299). I'll update with more info if I'm able to get the PrismLibrary to compile!

**Comment by brianlagunas** [comment]
This is the expected behavior. Prism does not persist parameters that were passed from page to page. You are responsible for passing parameters in a GoBackAsync call

**Comment by FrogsLegs** [comment]
@brianlagunas Thank you for looking into this, this is _not_ the default behavior of a standard UWP app. 
The source code of Prism also seems to attempt to pull these parameters, are you sure this is intended behavior? The concrete implementation of NavigationService inside for Windows 10 UWP checks the frame for any parameters:

`var previous = frameProvider.Frame.BackStack.Last().Parameter?.ToString();`

And then returns those parameters in the call to the `FrameFacade`:

```
return await _frameFacade.GoBackAsync(
                    parameters: parameters,
                    infoOverride: infoOverride);
```
Apologies if I'm missing something, but it seems strange Prism would remove what is a default behavior in UWP, and if so why does the source attempt to retrieve the params from the `BackStack.Last` entry?

**Comment by FrogsLegs** [comment]
Not sure of this is relevant, but we did migrate our project from Template10 to Prism, in Template10 the parameters were indeed passed into the view as expected when navigating back. We are only seeing this issue with Prism.

**Comment by dansiegel** [comment]
@FrogsLegs 100% sure... if you had context of the NavigationParameters from View3 when Navigating Back from View2 in a separate call that would 100% be a bug... Any state that you want to pass from one View to the next should be stateless between Navigation calls... meaning you are controlling it as part of that Navigation call. Hope that helps clarify it for you.

**Comment by totht91** [comment]
Prism should persist navigation parameters when go back action performs because it wouldn't be the same page with the same content. 

For example consider the following:
`ListPage -----some id----> DetailsPage (displays content based on the ID) ----(no param)----> SomePage`

If you go back from `SomePage`, `SomePage` can not, and should not know what was the ID what `DetailsPage` has got from `ListPage` but `DetailsPage` should display content based on the previous ID after go back operation.

PS: @brianlagunas You are so arogant these days and you close every issue without considering the problem a little bit. Rest a bit please.

**Comment by FrogsLegs** [comment]
@totht91 I thought I was going crazy after reading @dansiegel's comment.

This does not make sense that the parameter passed from View1 -> View2 (which gave View2 the ability to filter some data) is lost when then navigating from View3 back to View2.

**Comment by brianlagunas** [comment]
@totht91 Let make something very clear to everyone... I have absolutely zero obligation to anyone for anything. If I close something, I have a reason. That reason could be as simple as I don't feel like dealing with it. Your expectation that I am required to consider anything is laughable and a major problem with the attitude of the OSS community in general. All everyone does is just take and take and take. This is the exact attitude that has me considering leaving the OSS space altogether. I'm not arrogant, I'm tired, I'm frustrated, and I'm annoyed at how the OSS community treats its project maintainers.

I started working on Prism because it was fun.  It's no longer fun.

**Comment by FrogsLegs** [comment]
@brianlagunas It's clear that your contributions to the Prism library have been invaluable, and enabled folks like me to develop some awesome apps! I am grateful for your help on this issue and for making Prism what is is today.
That said, 
>  I don't feel like dealing with it
Really isn't the approach to handling issues within any application. I am more than happy to help resolve this, I'd love to get involved with an OS project and since Prism is the foundation for our app, let's go!
I'll try to get the required SDK's to compile the Prism library again over the weekend and see if I can make sense of this. I would ask however that you re-open this issue until it's at least established that this is a bug (or not).

Again thanks for your help!

**Comment by totht91** [comment]
@briannoyes Then it would be better to just leave the issues open and give the wheel to someone else. Closing issues with the reason ** I don't feel like dealing with it** does not help who opened the issue and does not help the library itself.

**Comment by brianlagunas** [comment]
Or better yet, I'll just shut the project down.

**Comment by bartlannoeye** [comment]
Going through the issue and the linked StackOverflow question again, I noticed there was some misunderstanding/miscommunication in this thread. I've removed off-topic replies so we can pick up this issue again for further investigation.

- Prism used to support storing parameters for forward navigation in v6

Example: Navigating from Page1 to Page2?param=true then to Page3. Going back to Page2, you would have param=true as parameter.

- Prism didn't support passing parameters for back navigation, neither does UWP, as Brian correctly mentions above.

@FrogsLegs seems to have an issue with the first case, which was supported in the past and might got lost when bringing in Template 10 code. We'll investigate this.
